source/adapters/level_zero/event.cpp

```diff
@@ -1373,9 +1386,10 @@ ur_result_t _ur_ze_event_list_t::createAndRetainUrZeEventList(
           const auto &ZeCommandList = CommandList->first;
           EventList[I]->RefCount.increment();
 
-          zeCommandListAppendWaitOnEvents(ZeCommandList, 1u,
-                                          &EventList[I]->ZeEvent);
-          zeEventHostSignal(MultiDeviceZeEvent);
+          ZE2UR_CALL(zeCommandListAppendWaitOnEvents,
+                     (ZeCommandList, 1u, &EventList[I]->ZeEvent));
+          if (!MultiDeviceEvent->CounterBasedEventsEnabled)
```

Comment:
The if here cannot be included in 0.9.x, this should be:
ZE2UR_CALL(zeEventHostSignal, (MultiDeviceZeEvent));

without the if.

Reply:
It might be I made a mistake while rebasing the PR itself, will fix that.